frontend/src/types/ResidentTypes.ts

```diff
@@ -1,10 +1,10 @@
 export type Resident = {
-  id?: number;
+  id: number;
   residentId?: string;
   initial: string;
   roomNum: number;
-  dateJoined: Date;
-  dateLeft?: Date;
+  dateJoined: string;
+  dateLeft?: string;
```

Comment:
I'm just thinking this might be a breaking change elsewhere in the app ; is this 100% required?